wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -1802,3 +1819,34 @@ fn supports_bgra8unorm_storage(
             && features3.contains(vk::FormatFeatureFlags2::STORAGE_WRITE_WITHOUT_FORMAT)
     }
 }
+
+// For https://github.com/gfx-rs/wgpu/issues/4599
+// Intel iGPUs with outdated drivers can break rendering if `VK_EXT_robustness2` is used.
+// Driver version 31.0.101.2115 works, but there's probably an earlier functional version.
+fn is_intel_igpu_outdated_for_robustness2(
+    props: vk::PhysicalDeviceProperties,
+    driver: Option<vk::PhysicalDeviceDriverPropertiesKHR>,
+) -> bool {
+    use crate::auxil::cstr_from_bytes_until_nul;
+
+    const DRIVER_VERSION_31_0_101: u32 = 0x194000;
+    const DRIVER_VERSION_WORKING: u32 = DRIVER_VERSION_31_0_101 + 2115;
```

Comment:
```suggestion
    const DRIVER_VERSION_WORKING: u32 = (101 << 14) | 2115; // X.X.101.2115
```

Not sure where the X.X is coming from but it's not reported by Vulkan.
Even Intel on [their page](https://www.intel.com/content/www/us/en/download/762755/intel-6th-10th-gen-processor-graphics-windows.html) only lists it as 101.2115 on the download button and release notes name.

Double checked the scheme [here](https://github.com/SaschaWillems/vulkan.gpuinfo.org/blob/4b05d1dcc4005c591e0e0ea901e1238d9ad0818c/includes/functions.php#L428).